packages/discord-utilities/src/lib/InteractionOptionResolver.ts

```diff
@@ -80,14 +80,16 @@ export class InteractionOptionResolver {
 		}
 	}
 
-	public get(name: string, required?: boolean | false): APIApplicationCommandInteractionDataOption | null;
-	public get(name: string, required: true): APIApplicationCommandInteractionDataOption;
-
 	/**
 	 * Gets an option by its name
 	 * @param name The name of the option
 	 * @param required Whether to throw an error if the option is not found
 	 */
+	public get<const Required extends boolean = false>(
```

Comment:
is `const` really needed?

Reply:
Yes, to make `get(true)` behave as `get(true as const)`.

Reply:
?
![Google Chrome Beta - 2024-11-02 at 17 00 28](https://github.com/user-attachments/assets/d2232dcb-373c-4e2c-ad13-44657f982357)